libvast/src/system/type_registry.cpp

```diff
@@ -98,17 +98,20 @@ caf::error type_registry_state::load_from_disk() {
 }
 
 void type_registry_state::insert(vast::type layout) {
-  // Flatten all incoming types first.
-  auto new_layout = flatten(layout);
-  auto& old_layouts = data[new_layout.name()];
+  auto& old_layouts = data[layout.name()];
   // Insert into the existing bucket.
-  auto [hint, success] = old_layouts.insert(std::move(new_layout));
+  auto [hint, success] = old_layouts.insert(std::move(layout));
   if (success) {
     // Check whether the new layout is compatible with the latest, i.e., whether
     // the new layout is a superset of it.
-    if (old_layouts.begin() != hint && is_subset(*old_layouts.begin(), *hint))
-      VAST_WARNING(self, "detected an incompatible version of", hint->name());
-    VAST_DEBUG(self, "registered", hint->name());
+    if (old_layouts.begin() != hint) {
+      if (!is_subset(*old_layouts.begin(), *hint))
+        VAST_WARNING(self, "detected an incompatible layout change for",
+                     hint->name());
+      else
+        VAST_VERBOSE(self, "detected a layout change for", hint->name());
+    }
+    VAST_INFO(self, "registered", hint->name());
```

Comment:
This is way too noisy on start with an empty db. We originally had this on verbose, and moved it to debug just recently. I'd say move the above "detected a layout change" from verbose to info and move this to debug again.

Reply:
Ah, I disabled the loading of the on-disk types during testing. Should have done a final check...